src/main2.js

```diff
@@ -0,0 +1,166 @@
+const buttonLeft = document.querySelector('.button--left');
+const buttonRight = document.querySelector('.button--right');
+
+//Switch
+
+const change = document.querySelector('.switch');
+
+change.addEventListener('click', () => {
+    change.firstElementChild.classList.toggle('point--active');
+    buttonLeft.classList.toggle('button--hidden');
+    buttonRight.classList.toggle('button--hidden');
+});
+
+//Carousel
+
+const dots = document.querySelectorAll('.dotsNav__dot');
+const carousel = document.querySelector('.carousel__container');
+
+window.addEventListener('load', () => {
+    let slides = document.querySelectorAll('.carousel__slide');
+    let elementWidth = slides[0].offsetWidth;
+    let isInTransition = false;
+    //Middle Dot
+    let activeDot = 3;
+
+    const moveCarousel = (slides, elementWidth) => {
+        isInTransition = true;
+        slides.forEach((slide, index) => {
+            slide.style.left = `${elementWidth * (index - 1)}px`;
+        });
+    };
+
+    moveCarousel(slides, elementWidth);
+
+    slides.forEach(slide => {
+        slide.animate(
+            [
+                // keyframes
+                { opacity: '0' },
+                { opacity: '1' }
+            ],
+            {
+                // timing options
+                duration: 500,
+                delay: 400,
+                iterations: 1,
+                fill: 'both'
+            }
+        );
+    });
+
+    const setActiveElement = slides => {
+        const activeElement = slides[3];
+
+        carousel
+            .querySelector('.carousel__slide--active')
+            .classList.remove('carousel__slide--active');
+        //The middle element
+        activeElement.classList.add('carousel__slide--active');
+
+        document
+            .querySelector('.dotsNav__dot--active')
+            .classList.remove('dotsNav__dot--active');
+        dots[activeDot].classList.add('dotsNav__dot--active');
+    };
+
+    const moveToRight = () => {
+        if (isInTransition === true) return;
+        carousel.appendChild(carousel.firstElementChild);
+        slides = carousel.querySelectorAll('.carousel__slide');
```

Comment:
może do osobnej funkcji:
```js
getSlides(carousel) {
     let element  = carousel;
     element.appendChild(carousel.firstElementChild);
     return element.querySelectorAll('.carousel__slide')
}
```
wtedy nie mutujesz elementów (ale nie wiem czy to tak ma działać u Ciebie)